site/content/api/Credentials-API.md

```diff
@@ -0,0 +1,316 @@
+| *action*         | yes       | *application-properties* | UTF-8 *string*  | MUST contain the value `add`.       |
+
+The request message MUST include payload as defined in [Request Payload]({{< relref "#request-payload" >}}).
+
+**Response Message Format**
+
+A response to an *add credentials* request only contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message's body is empty. 
+
+The response message's *status* property may contain the following codes:
+
+| Code  | Description |
+| :---- | :---------- |
+| *201* | Created, the credentials have been added successfully. |
+| *409* | Conflict, there already exist credentials with the `type` and `auth-id` provided in the request payload for the *device_id*. |
+| *412* | Precondition Failed, there is no device registered with the given *device_id* within the *tenant_id*. |
+
+For status codes indicating an error (codes in the `400 - 499` range) the message body MAY contain a detailed description of the error that occurred.
+
+## Get Credentials
+
+Protocol adapters use this command to *look up* credentials of a particular type for a device identity.
+
+**Message Flow**
+
+*TODO* add sequence diagram
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on an *get credentials* message in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name             | Mandatory | Location                 | Type            | Description                   |
+| :--------------- | :-------: | :----------------------- | :-------------- | :---------------------------- |
+| *action*         | yes       | *application-properties* | UTF-8 *string*  | MUST contain the value `get`. |
+
+The body of the message MUST consist of a single *AMQP Value* section containing a UTF-8 encoded string representation of a single JSON object as follows:
+
+~~~json
+{
+  "type": "${credential_type}",
+  "auth-id": "${authentication_identifier}"
+}
+~~~
+
+The `type` property MUST contain the exact type of credentials to look up. The `auth-id` property MUST contain the authentication identifier to look up credentials for.
+
+**Response Message Format**
+
+A response to a *get credentials* request only contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message includes payload as defined in [Response Payload]({{< relref "#response-payload" >}}).
+
+The response message's *status* property may contain the following codes:
+
+| Code  | Description |
+| :---- | :---------- |
+| *200* | OK, the payload contains the credentials for the authentication identifier. |
+| *404* | Not Found, there are no credentials registered matching the critera. |
+
+For status codes indicating an error (codes in the `400 - 499` range) the message body MAY contain a detailed description of the error that occurred.
+
```

Comment:
I am currently adding a `enabled`, `not-before` and `not-after` to the request payload definition.
However, while doing so I started to wonder if and how we would/could enforce their usage in e.g. protocol adapters. In the end, it is up to the protocol adapter whether it wants to authenticate the device (based on the credentials returned by the Credentials API) or not. Since we already assume a *trust relationship* between protocol adapters and the Hono server, I guess that is a risk we need to take. Otherwise we could fail the look up of credentials that are not *enabled* or *valid* at the moment. WDYT?